src/main/java/org/gitlab4j/api/SearchApi.java

```diff
@@ -226,29 +246,58 @@ public SearchApi(GitLabApi gitLabApi) {
     }
 
     /**
      * Search within the specified project.  If a user is not a member of a project and the project is private,
      * a request on that project will result to a 404 status code.
      *
-     * <pre><code>GitLab Endpoint: POST /project/:projectId/search?scope=:scope&amp;search=:search-query</code></pre>
+     * <pre><code>GitLab Endpoint: POST /projects/:projectId/search?scope=:scope&amp;search=:search-query&amp;ref=ref</code></pre>
      *
      * @param projectIdOrPath the project in the form of an Integer(ID), String(path), or Project instance, required
      * @param scope search the expression within the specified scope. Currently these scopes are supported:
      *               issues, merge_requests, milestones, notes, wiki_blobs, commits, blobs, users
      * @param search the search query
+     * @param ref the name of a repository branch or tag to search on. The project’s default branch is used by 
+     *             default. This is only applicable for scopes: commits, blobs, and wiki_blobs.
+     * @return a Stream containing the object type specified by the scope
+     * @throws GitLabApiException if any exception occurs
+     * @since GitLab 10.5
+     */
+    public Stream<?> projectSearchStream(Object projectIdOrPath, ProjectSearchScope scope, String search, String ref) throws GitLabApiException {
+        return (projectSearch(projectIdOrPath, scope, search, ref, getDefaultPerPage()).stream());
+    }
+
+    /**
+     * Search within the specified project.  If a user is not a member of a project and the project is private,
+     * a request on that project will result to a 404 status code.
+     *
+     * <pre><code>GitLab Endpoint: POST /project/:projectId/search?scope=:scope&amp;search=:search-query&amp;ref=ref</code></pre>
+     *
+     * @param projectIdOrPath the project in the form of an Integer(ID), String(path), or Project instance, required
+     * @param scope search the expression within the specified scope. Currently these scopes are supported:
+     *               issues, merge_requests, milestones, notes, wiki_blobs, commits, blobs, users
+     * @param search the search query
+     * @param ref the name of a repository branch or tag to search on. The project’s default branch is used by 
+     *             default. This is only applicable for scopes: commits, blobs, and wiki_blobs.
      * @param itemsPerPage the number of items that will be fetched per page
      * @return a Pager containing the object type specified by the scope
      * @throws GitLabApiException if any exception occurs
      * @since GitLab 10.5
      */
-    public Pager<?> projectSearch(Object projectIdOrPath, ProjectSearchScope scope, String search, int itemsPerPage) throws GitLabApiException {
+    public Pager<?> projectSearch(Object projectIdOrPath, ProjectSearchScope scope, String search, String ref, int itemsPerPage) throws GitLabApiException {
```

Comment:
W still need a method that does not take a ref:
```
 public Pager<?> projectSearch(Object projectIdOrPath, ProjectSearchScope scope, String search, int itemsPerPage) throws GitLabApiException
```

Reply:
Hi @gmessner, 

Effectively I should have kept the original method in addition to the new one (as I did for the 2 other methods). I just added a new commit to correct that.

Olivier